mne/inverse_sparse/mxne_inverse.py

```diff
@@ -161,11 +162,45 @@ def _make_sparse_stc(X, active_set, forward, tmin, tstep,
 
 
 @verbose
+def _make_dipoles(X, active_set, forward, tmin, tstep, M, M_estimated,
+                  verbose=None):
+
+    times = np.arange(tmin, tmin + X.shape[1] * tstep, tstep)
+
+    active_idx = np.where(active_set)[0]
+    n_dip_per_pos = 1 if is_fixed_orient(forward) else 3
+    if n_dip_per_pos > 1:
+        active_idx = np.unique(active_idx // n_dip_per_pos)
+
+    gof = linalg.norm(M_estimated, axis=0) / linalg.norm(M, axis=0) * 100.
+
+    dipoles = []
+    for k, i_dip in enumerate(active_idx):
+        i_pos = forward['source_rr'][i_dip][np.newaxis, :].repeat(len(times),
+                                                                  axis=0)
+        X_ = X[k * n_dip_per_pos: (k + 1) * n_dip_per_pos]
+        if n_dip_per_pos == 1:
+            amplitude = X_
+            i_ori = forward['source_nn'][i_dip]
+        elif forward['surf_ori']:
+            X_ = np.dot(forward['source_nn'][i_dip *
+                        n_dip_per_pos:(i_dip + 1) * n_dip_per_pos].T, X_)
+            amplitude = linalg.norm(X_, ord=None, axis=0)
+            i_ori = (X_ / amplitude).T
+        else:
+            amplitude = linalg.norm(X_, ord=None, axis=0)
+            i_ori = (X_ / amplitude).T
+        dipoles.append(Dipole(times, i_pos, amplitude, i_ori, gof))
```

Comment:
dipole orientation can vary between time samples, i_ori is activation matrix (3 dipoles per location) normalized to 1 and rotated to xyz coordinate system